### ISSUE TITLE
Bug in our context building for non-instance methods

### ISSUE THREAD

**Issue by troessner** [troessner/reek#793]
#724 uncovered a bug in our context building in regards to non-instance methods.

The following spec illustrates this bug:

```Ruby
    it "properly recognizes singleton_methods correctly" do
      src = <<-EOS
        class Car
          class << self
            def drive; end
          end

          def self.start; end
        end
      EOS

      syntax_tree = Reek::Source::SourceCode.from(src).syntax_tree

      context_tree = Reek::ContextBuilder.new(syntax_tree).send(:context_tree)

      class_node = context_tree.children.first
      drive_method = class_node.children.first
      start_method = class_node.children.last

      expect(start_method.class).to eq(Reek::Context::SingletonMethodContext)
      expect(drive_method.class).to eq(Reek::Context::SingletonMethodContext)
    end
```

Both methods should be recognized as SingletonMethod, but only the first one is:

```
Failures:

  1) Reek::ContextBuilder#initialize properly recognizes singleton_methods correctly
     Failure/Error: expect(drive_method.class).to eq(Reek::Context::SingletonMethodContext)

       expected: Reek::Context::SingletonMethodContext
            got: Reek::Context::MethodContext

       (compared using ==)
```

I thought we had fix this bug a looooong time ago but apparently not.

**Comment by mvz** [comment]
I'm picking this up as part of #792.

**Comment by mvz** [comment]
(I think we're checking if some method is a singleton method more dynamically, actually, due to `module_function`).

**Comment by troessner** [comment]
Wait, why are you closing this? Thats confusing since it implies that the fix is in master already. 
But right now the PR is not even in reviewable state...?

**Comment by mvz** [comment]
It was autoclosed by GitHub because #794 used the word 'fix' and this ticket number in the same sentence :smile:.

**Comment by troessner** [comment]
Ah, i see :laughing: 
Reopening this one!

**Comment by mvz** [comment]
Fixed in #796.
